app/src/main/java/io/github/androidacademyglobal/profile/ProfileScreen.kt

```diff
@@ -0,0 +1,141 @@
+import androidx.compose.foundation.ScrollState
+import androidx.compose.foundation.layout.BoxWithConstraints
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.heightIn
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.Divider
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.TopAppBarDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.input.nestedscroll.nestedScroll
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import io.github.androidacademyglobal.R
+import io.github.androidacademyglobal.components.baselineHeight
+import io.github.androidacademyglobal.ui.theme.MARGIN_LARGE
+import io.github.androidacademyglobal.ui.theme.MARGIN_MEDIUM
+import io.github.androidacademyglobal.ui.theme.MARGIN_SMALL
+import io.github.androidacademyglobal.ui.theme.PROFILE_SCREEN_BOTTOM_MARGIN
+import io.github.androidacademyglobal.ui.theme.ZERO
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun ProfileScreen(userData: ProfileScreenState) {
+
+    val scrollState = rememberScrollState()
+    val scrollBehavior = remember { TopAppBarDefaults.pinnedScrollBehavior() }
+
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .nestedScroll(scrollBehavior.nestedScrollConnection)
+    ) {
+        BoxWithConstraints(modifier = Modifier.weight(1f)) {
+            Surface {
+                Column(
+                    modifier = Modifier
+                        .fillMaxSize()
+                        .verticalScroll(scrollState),
+                ) {
+                    ProfileHeader(
+                        scrollState,
+                        userData,
+                        this@BoxWithConstraints.maxHeight
+                    )
+                    UserInfoFields(userData, this@BoxWithConstraints.maxHeight)
+                }
+            }
+        }
+    }
+}
+
+@Composable
+private fun ProfileHeader(
+    scrollState: ScrollState,
+    data: ProfileScreenState,
+    containerHeight: Dp
+) {
+    val offset = (scrollState.value / 2)
```

Comment:
Generally, looks good I think, though I'd assume a bit more detailed implementation (starting from a repository and domain model in `core` module and via ViewModel getting this data to views)